### PR TITLE
feat: the very emotional article

### PR DESCRIPTION
**Pull request by EthanStandel**

### CODE REVIEW

File: src/pages/blog/optimizing-emotion.mdx [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+## Just how much work is Emotion doing?
+
+Just like at the build time of a CSS preprocessor, Emotion has to parse the style passed into it. Emotion _does not_ simply pass its input strings into a `<style>` tag, it must parse the style strings into an object tree. The reason your style strings must be parsed by Emotion before the browser is because Emotion offers SCSS-like nesting syntax as it uses [`stylis`](https://www.npmjs.com/package/stylis) as it's underlying stylsheet language.
```
Comment (nring) [inline src/pages/blog/optimizing-emotion.mdx]: TIL about Stylis

File: src/pages/blog/optimizing-emotion.mdx [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+## How can we optimize our use of Emotion in React applications?
+
+### Opimization I. Avoid doing construction of styles in render functions.
+
+Now that it's clear how much work Emotion has to do at runtime, it becomes far more obvious that we want to do this work as infrequently as possible. So this means avoiding using Emotion style constructions in renders as much as we possibly can. Despite almost every example in [the Emotion documentation](https://emotion.sh/docs/introduction) showing style constructions in render functions, this is not the best practice. The documentation, to me, reads as a bit too focussed on what is possible with Emotion and not enough on what would be best practice for performance.
```
Comment (nring) [inline src/pages/blog/optimizing-emotion.mdx]: I thought `focussed` needed to be one `s` but apparently both are correct!

File: src/pages/blog/optimizing-emotion.mdx [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+import * as React from 'react';
+import { css } from '@emotion/react';
+
+export const MyButton = (
+  props: React.DetailedHTMLProps<
+    React.ButtonHTMLAttributes<HTMLButtonElement>,
+    HTMLButtonElement
+  >
+) => (
+  <button
```
Comment (nring) [inline src/pages/blog/optimizing-emotion.mdx:56]: I didn't know that this was so inefficient but I still hated it. Trying to parse changes to either JSX or styles in a PR is so much difficult when they're co-located like this.

File: src/pages/blog/optimizing-emotion.mdx [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+
+However, this plugin is not always an option. If you're not using Babel or a framework/build-tool that suppports Babel plugins, then you'll be entirely without access to this optimization. This leads back to my advice that it is better write code that is perfomant from the start, rather than relying on a plugin to fix it for you. _Technically_ we are still relying on the JavaScript build toolchain to minify our code, but this is a much more common and reliable optimization than relying on a specific plugin to minify template string styles.
+
+### Opimization III. Consider if you really need Emotion and everything that it offers 🤷‍♂️
+
+The final solution to optimizing use of Emotion in React applications that I would like to propose for consideration is: not using it. There are many situations where you may have to pull in Emotion. If your organization has standardized use of a particular component library which relies on Emotion, like MUI or Chakra, then you already depend on Emotion. You are best off just using Emotion because the tool is already in front of you. So use it, and use it well with the above recommended optimizations.
```
Comment (nring) [inline src/pages/blog/optimizing-emotion.mdx]: ❤️

File: src/pages/blog/optimizing-emotion.mdx [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+## Conclusion
+
+Emotion is a well engineered, unopinionated, and surprsingly performant ecosystem of tools that may offer the optimal implementation of a sensible solution for a complex problem, but it is not without it's drawbacks. It is important to understand the tradeoffs that you are making when you choose to use Emotion or any styling system in your application or library. It is also important to understand that there are many ways to optimize your use of Emotion, and the docs aren't always up-front about it and seem to focus more directly on the enhanced capabilities and developer experience than the potential drawbacks. If you come into a project with all this awareness in mind and are able to account for optimizations slightly earlier, than you can avoid [having to do a lot of refactoring later on if Emotion starts to feel like it's not meeting your needs](https://dev.to/srmagura/why-were-breaking-up-wiht-css-in-js-4g9b).
```
Comment (nring) [inline src/pages/blog/optimizing-emotion.mdx]: Haha I was waiting for a link to this article!

**Review by nring — COMMENTED** [review]
This is really well written. I only have a few minor comments but nicely done!

File: src/pages/blog/optimizing-emotion.mdx (hunk 1 above)
Comment (EthanStandel) [reply src/pages/blog/optimizing-emotion.mdx]: I learned about it this week. I did a lot of deep-diving into the source contributor docs of Emotion to write this article 😅

File: src/pages/blog/optimizing-emotion.mdx (hunk 2 above)
Comment (EthanStandel) [reply src/pages/blog/optimizing-emotion.mdx]: You know I thought I did spell it wrong at a point and I just Google'd "focussed" and when it didn't say "Did you mean 'focused'?" I assumed I was right. But evidently, this is the British/Australian spelling so I'm going to swap it to one S for consistency.

File: src/pages/blog/optimizing-emotion.mdx (hunk 3 above)
Comment (EthanStandel) [reply src/pages/blog/optimizing-emotion.mdx:56]: Tbh, I actually don't hate it and wouldn't mind using this style in a SolidJS app where you wouldn't ever have to worry about unassociated rerenders.

But at this point, I'm kind of an all-in Tailwind convert for new apps (thanks David)

File: src/pages/blog/optimizing-emotion.mdx (hunk 5 above)
Comment (EthanStandel) [reply src/pages/blog/optimizing-emotion.mdx]: I intended to sneak it in earlier, but it felt the most sensible to put it here as it kind of seems like the overarching consequential risk of leaving Emotion code unopimized.

**Pull request merged**